samples/sdl_ttf/main.c

```diff
@@ -0,0 +1,157 @@
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/xbox.h>
+#include <hal/video.h>
+#include <SDL.h>
+#include <SDL_ttf.h>
+
+// SDL requires this symbol to exist
+void _exit() {
+}
+
+// Shut down initialized systems
+void cleanup(int lvl) {
+  if(lvl >= 3) {
+    TTF_Quit();
+  }
+  if(lvl >= 2) {
+    SDL_Quit();
+  }
+  if (lvl >= 1) {
+    pb_kill();
+  }
+}
+
+
+void main() {
+  const extern int SCREEN_WIDTH;
+  const extern int SCREEN_HEIGHT;
+
+  int init_ret;
+  int init_level = 0;
+
+  init_ret = pb_init();
+  if (init_ret != 0) {
+    debugPrint("pb_init() failed!");
+    XSleep(2000);
+    return;
+  }
+  pb_show_debug_screen();
+
+  ++init_level;
+  
+  init_ret = SDL_VideoInit(NULL);
+  if (init_ret != 0) {
+    debugPrint("SDL_VideoInit failed: %s", SDL_GetError());
+    XSleep(2000);
+    cleanup(init_level);
+    return;
+  }
+  ++init_level;
+
+  init_ret = TTF_Init();
+  if (init_ret != 0) {
+    debugPrint("TTF_Init failed: %s", TTF_GetError());
+    XSleep(2000);
+    cleanup(init_level);
+    return;
+  }
+  ++init_level;
+
+  SDL_Window   *window;
+  SDL_Renderer *renderer;
+  SDL_Surface  *surface;
+  SDL_Texture  *texture;
+  TTF_Font     *font;
+
+  SDL_Color     textcolor = {0x7F, 0xFF, 0x7F, 0xFF};
+
+  window = SDL_CreateWindow("NXDK SDL_ttf sample",
+                            SDL_WINDOWPOS_UNDEFINED,
+                            SDL_WINDOWPOS_UNDEFINED,
+                            SCREEN_WIDTH, SCREEN_HEIGHT,
+                            SDL_WINDOW_SHOWN);
+  if (window == NULL) {
+    debugPrint("Window creation failed: %s", SDL_GetError());
+    XSleep(2000);
+    cleanup(init_level);
+    return;
+  }
+
+  renderer = SDL_CreateRenderer(window, -1, 0);
+  if (renderer == NULL) {
+    debugPrint("CreateRenderer failed: %s\n", SDL_GetError());
+    XSleep(2000);
+    SDL_DestroyWindow(window);
+    cleanup(init_level);
+    return;
+  }
+
+  font = TTF_OpenFont("vegur-regular.ttf", 96);
+  if (font == NULL) {
+    debugPrint("Couldn't load font: %s", TTF_GetError());
+    XSleep(2000);
+    return;
+  }
+
+  surface = TTF_RenderText_Blended(font, "NXDK", textcolor);
+  TTF_CloseFont(font);
+  if (surface == NULL) {
+    debugPrint("TTF_RenderText failed: %s", TTF_GetError());
+    XSleep(2000);
+    SDL_DestroyRenderer(renderer);
+    SDL_DestroyWindow(window);
+    cleanup(init_level);
+    return;
+  }
+
+  texture = SDL_CreateTextureFromSurface(renderer, surface);
+  SDL_FreeSurface(surface);
+  if (texture == NULL) {
+    debugPrint("Couldn't create texture: %s\n", SDL_GetError());
+    XSleep(2000);
+    SDL_DestroyRenderer(renderer);
+    SDL_DestroyWindow(window);
+    cleanup(init_level);
+    return;
+  }
+
+  SDL_Rect outputPosition;
+  SDL_QueryTexture(texture, NULL, NULL, &outputPosition.w, &outputPosition.h);
+  SDL_SetRenderDrawColor(renderer, 0x00, 0x00, 0x00, 0xFF);
```

Comment:
I'd move this into the renderer loop; if someone adds other draw calls this would otherwise break.

It's also better to put related code together. I'd also assume all of these calls are very cheap (performance wise).